### ISSUE TITLE
[DatePicker] a11y issue with selected state of Year and Date buttons

### ISSUE THREAD

**Issue by slayybelle** [mui/mui-x#9916]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

The visually selected Date and Year lack any attributes which make is clear to screen readers that this button is in a selected state.  Auditors recommend setting the button to aria-pressed="true" or "false" to indicate its state.

### Expected behavior 🤔

UI components such as Date and Year buttons should convey their selected/not selected state to screen readers. 
Pls add aria-pressed="true" or "false"

### Steps to reproduce 🕹

Steps:

1. Go to Date Picker demo, Basic Usage:  https://mui.com/components/date-picker/#basic-usage
2. using Devtools, choose the current date that is selected:
<img width="1544" alt="Screen Shot 2022-02-15 at 10 00 52 AM" src="https://user-images.githubusercontent.com/32496958/154121543-d1517788-035e-40fb-9b97-9c80499d55e2.png">

3. Note under Element pane in devtools that there is no aria attribute indicating the selected state of the button
4. using Devtools, choose the current year that is selected:
<img width="1478" alt="Screen Shot 2022-02-15 at 10 02 31 AM" src="https://user-images.githubusercontent.com/32496958/154121773-3c0d9081-88d2-4eb8-9d93-4770d9f4b7de.png">

5. Note under Element pane in devtools that there is no aria attribute indicating the selected state of the button

### Context 🔦

W3C ARIA Recommendation: https://www.w3.org/TR/wai-aria-1.1/#aria-pressed

### Your environment 🌎

using the demo page

**Comment by LukasTy** [comment]
I've done some exploration.
Let's first begin by prefacing that VoiceOver is the worst screen reader there is and it has endless issues, we are not going to target it as the platform we want to optimize. Our current a11y target platform is NVDA on Windows as it and JAWS are the most popular and feature-rich screen readers on the market.

IMHO, the current implementation of `DateCalendar` is working correctly on NVDA, it's telling the selected and not selected days when navigating.
What we are missing is the replication of the same logic on both `MonthCalendar` as well as `YearCalendar`.
Both of those do not currently have the same role structure as the `DateCalendar`, which works as a grid with grid cells.
Because `Month` and `Year` calendars do not have the grid role structure, adding `aria-pressed` on those would have a positive effect, but I'd say that we should lean towards having similar implementation and refactor those calendars to have grid cells, in which case, the `aria-pressed` attribute would no longer have any effect and we'd need to use the https://www.w3.org/TR/wai-aria-1.1/#aria-selected as per the spec.

Side note, reading both `aria-pressed` and `aria-checked` descriptions does give a somewhat clear definition that those attributes are not the most correct given how the calendars work.
